assets/js/data/cart/selectors.js

```diff
@@ -106,3 +106,30 @@ export const isRemovingCoupon = ( state ) => {
 export const getCouponBeingRemoved = ( state ) => {
 	return state.metaData.removingCoupon || '';
 };
+
+/**
+ * Returns cart item matching specified key.
+ *
+ * @param {Object} state The current state.
+ * @param {string} cartItemKey Key for a cart item.
+ * @return {Object} Cart item object, or undefined if not found.
+ */
+export const getCartItem = ( state, cartItemKey ) => {
+	return state.cartData.items.find( cartItem => ( cartItem.key === cartItemKey ) );
+};
+
+/**
+ * Returns true if the quantity is being updated for the specified cart item.
+ *
+ * @param {Object} state The current state.
+ * @param {string} cartItemKey Key for a cart item.
+ * @return {boolean} True if a item has a pending request to delete / update quantity.
+ */
+export const isItemQuantityPending = ( state, cartItemKey ) => {
+	const cartItem = getCartItem( state, cartItemKey );
+	if ( ! cartItem ) {
+		return false;
+	}
+
+	return !! cartItem.isQuantityPending;
+};
```

Comment:
So I do see you are accounting for whether the `cartItem` has the `isQuantityPending` key here. But related to my comments for the reducer, I don't think this should be a property of the cartItem (as it pollutes the shape returned from the api). If you follow my suggestion in the reducer then you likely can use `Array.some` to check if the key is in the state.